tests/Unit/ActionTesting.hpp

```diff
@@ -926,4 +946,61 @@ class MockRuntimeSystem {
   Inboxes inboxes_;
   TupleOfMockDistributedObjects local_algorithms_;
 };
+
+/// Returns a vector of all the indices of the Components
+/// in the ComponentList that have queued actions.
+template <typename ComponentList, typename MockRuntimeSystem>
+std::vector<size_t> indices_of_components_with_queued_actions(
+    const gsl::not_null<MockRuntimeSystem*> runner) noexcept {
+  std::vector<size_t> result{};
+  size_t i = 0;
+  tmpl::for_each<ComponentList>([&](auto tag) noexcept {
+    using Tag = typename decltype(tag)::type;
+    // Passing 0 as the array_index.  Intended for use with
+    // singleton components, which need an index passed because the
+    // mocking framework assumes every component is an array.
+    if (not runner->template is_simple_action_queue_empty<Tag>(0)) {
+      result.push_back(i);
+    }
+    ++i;
+  });
+  return result;
+}
+
+/// \cond
+namespace detail {
+// Helper function used in invoke_queued_simple_action.
+template <typename ComponentList, typename MockRuntimeSystem, size_t... Is>
+void invoke_queued_action(const gsl::not_null<MockRuntimeSystem*> runner,
+                          const size_t component_to_invoke,
+                          std::index_sequence<Is...> /*meta*/) noexcept {
+  const auto helper = [ component_to_invoke, &runner ](auto I) noexcept {
+    if (I.value == component_to_invoke) {
+      // Passing 0 as the array_index.  Intended for use with
+      // singleton components, which need an index passed because the
+      // mocking framework assumes every component is an array.
+      runner->template invoke_queued_simple_action<
+          tmpl::at_c<ComponentList, I.value>>(0);
+    }
+  };
+  EXPAND_PACK_LEFT_TO_RIGHT(helper(std::integral_constant<size_t, Is>{}));
+}
+}  // namespace detail
+/// \endcond
+
+/// Invokes the next queued action on a random Component.
```

Comment:
If this will only work with singleton components that should be documented

Reply:
Ok, I'm doing better.  It will now take an array_index so it should work with any mocked component.